src/graph.py

```diff
@@ -0,0 +1,275 @@
+from time import sleep, time_ns
+
+import ccxt  # type: ignore
+
+
+class PricePath:
+    def __init__(self, exchanges: list = [], gdict: dict = {}, cache: dict = {}):
+        if not exchanges:
+            exchanges = ["binance", "coinbasepro"]
+        self.gdict = gdict
+        self.cache = cache
+        self.priority: dict[str, int] = {}
+        # saves the priority for a certain path so that bad paths can be skipped
+        allpairs = []
+
+        for exchange_id in exchanges:
+            exchange_class = getattr(ccxt, exchange_id)
+            exchange = exchange_class()
+            markets = []
+            markets = exchange.fetch_markets()
+            if exchange.has["fetchOHLCV"]:
+
+                allpairs.extend(
+                    [(i["base"], i["quote"], exchange_id, i["symbol"]) for i in markets]
+                )
+            else:
+                print(
+                    f"{exchange.name} Does not support fetch ohlcv. ignoring exchange and {len(markets)} pairs."
+                )
+        allpairs = list(set(allpairs))  # fast an easy deduplication
```

Comment:
actually i don't know. The only thing i know that list(set(list)) is one of the fastest way to deduplicate a unordered list (an probably the cleanest), because of the uniqueness of the set but i havent tested if set directly is faster

Reply:
> Is it faster to initialie allpairs as set? Instead of running `list(set(...))` afterwards?

set is an immutable object hence this is not possible

Reply:
I don't think so. A set might be unordered but can definitely be used like a list.

Reply:
Sry i had my Information wrong ^^ i had in my head that Sets are immutable and totally forgot that Set.add exists